sdks/java/harness/src/main/java/org/apache/beam/fn/harness/data/BeamFnDataClient.java

```diff
@@ -27,37 +27,37 @@
 import org.apache.beam.sdk.util.WindowedValue;
 
 /**
- * The {@link BeamFnDataClient} is able to forward inbound elements to a consumer and is also a
- * consumer of outbound elements. Callers can register themselves as consumers for inbound elements
- * or can get a handle for a consumer for outbound elements.
+ * The {@link BeamFnDataClient} is able to forward inbound elements to a {@link FnDataReceiver} and
+ * is also a receiver of outbound elements. Callers can register themselves as receivers for inbound
```

Comment:
Its not that BeamFnDataClient is a receiver but knows how to construct one or return a cached one. Maybe not worth trying to wordsmith this comment though.